From f0d702d0bcec96920b3b61868ee8dfd9732d9ac1 Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Thu, 10 Jan 2019 21:29:55 +0000
Subject: [PATCH 6/9] StatusBarSettings: Hide network traffic settings if
 device has a notch

 * Now a must, since the view is horizontally centered.

Change-Id: Iab6f057e5626ce5c3d13ce8041767b3c04b6d23b
---
 .../lineageparts/statusbar/StatusBarSettings.java   | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java b/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
index 9f4fb6d..e452dc0 100644
--- a/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
+++ b/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2014-2015 The CyanogenMod Project
- *               2017-2018 The LineageOS Project
+ *               2017-2019 The LineageOS Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -51,6 +51,8 @@ public class StatusBarSettings extends SettingsPreferenceFragment
     private static final int PULLDOWN_DIR_RIGHT = 1;
     private static final int PULLDOWN_DIR_LEFT = 2;
 
+    private static final String NETWORK_TRAFFIC_SETTINGS = "network_traffic_settings";
+
     private LineageSystemSettingListPreference mQuickPulldown;
     private LineageSystemSettingListPreference mStatusBarClock;
     private LineageSystemSettingListPreference mStatusBarAmPm;
@@ -65,6 +67,15 @@ public class StatusBarSettings extends SettingsPreferenceFragment
         super.onCreate(savedInstanceState);
         addPreferencesFromResource(R.xml.status_bar_settings);
 
+        final boolean hasNotch = getResources().getBoolean(
+                org.lineageos.platform.internal.R.bool.config_haveNotch);
+
+        if (hasNotch) {
+            PreferenceScreen networkTrafficPreferenceScreen =
+                    (PreferenceScreen) findPreference(NETWORK_TRAFFIC_SETTINGS);
+            getPreferenceScreen().removePreference(networkTrafficPreferenceScreen);
+        }
+
         mStatusBarAmPm =
                 (LineageSystemSettingListPreference) findPreference(STATUS_BAR_AM_PM);
         mStatusBarClock =
-- 
2.17.1

